### ISSUE TITLE
What is the mod all about?

### ISSUE THREAD

**Issue by LordDennis18** [BlueThecno/Convenience-Mod#1]
Is it only for upgraded structures? Like conveyors and factories?

**Comment by BlueTechnoCat** [comment]
Pretty much, is mostly for adding those things that sound interesting and can be added with a mod rather than having the usual people screaming at you in the Mindustry-Suggestions Repository

### Note

It's mostly for new blocks that works with the existing items and liquids in the game, which means no new items or liquids

**Comment by LordDennis18** [comment]
Routers and Junctions

•Titanium router and junction- much faster distribution than a normal router and junction.

•Plastanium router and junction- much faster distribution than titanium router and junction.

•Solar panel router- not only it distributes items it can also produce a little amount of power. Much slower than a titanium router.

•Phase Fabric router and junction- much quicker distribution than plastanium router and junction. Can also deflect bullets.

•Surge Alloy router and junction- the quickest distributor and most expensive router and junction. Can also cast a small chance of lightning

•Omega Distributor- a 4x4 distributor that requires power to function. Distributes quicker than plastanium router.

•Router Randomizer- randomizes an item when getting distributed. Vanilla and modded items works on it.

Example: a copper went to the router randomizer it may distribute lead, metaglass, sand or modded items.

(Same with the conduits and liquid routers so the suggestion is not that long.)

Drills
•Excavator- Drills beneath the ground to excavate all kinds of vanilla items but on a slower production and requires power. 
 
•Surge Alloy Excavator- A massive drill that requires a huge amount of power and only produces surge alloy every 7-10 seconds

•Slag Extractor- drills deep beneath the underground to extract slag. Requires power. 

•Cryofluid Extractor- drills beneath the snow or ice to extract cryofluid takes a while to extract. Doesn't require power since it takes power from the sun.

Walls
• Shielded walls- from copper to surge alloy walls that now has a shield, much expensive walls and requires power.

•Slag wall- when destroyed it spills out slag and burning enemies who step on it.

•Tar Wall- slows enemies down when destroyed.

Traps

•Pit trap- instantly kills t1 and t2 from falling does not affect flying and spooder enemies.

•E.M.P mine- when detonated it stuns enemies for a bit.

•Bear trap- stuns and continues to damage the enemy for a short time.

•Oil Barrel- spills out oil and immediately spreads flames.

•Flash mine- creates a flash and confuses enemies for a short time.

**Comment by BlueTechnoCat** [comment]
**O_O**

**Comment by BlueTechnoCat** [comment]
Wait, The Surge Alloy Excavator extracts Surge from the floor right?

Also I will have to test some of this stuff in the case of the routers since, to my knowledge at least, routers are pretty much instant, they don't teleport items because of backflow

**Comment by LordDennis18** [comment]
> Wait, The Surge Alloy Excavator extracts Surge from the floor right?
> 
> Also I will have to test some of this stuff in the case of the routers since, to my knowledge at least, routers are pretty much instant, they don't teleport items because of backflow

Yes, it digs deep beneath the floor but takes a while to excavate surge alloy since it's very rare. And maybe it can only dig in stone, sand or dark sand it depends on the excavator.

Also add an improved version of the surge alloy excavator but it's more expensive.
